src/server.cpp

```diff
@@ -37,44 +35,45 @@ void Server::handle_receive(const error_code& error, std::size_t bytes)
         return;
     }
 
-    size_t length = recv_buffer_[2] * 256 + recv_buffer_[3];
+    try
+    {
+        Packet packet = makeResponse(Packet(m_recvBuffer, bytes));
+
+        m_socket.async_send_to(boost::asio::buffer(packet.makeSendBuffer("secret"), 20),
+            m_remoteEndpoint, std::bind(&Server::handleSend, this, pls::_1, pls::_2));
 
-    if (bytes < length)
+        startReceive();
+    }
+    catch (size_t length)
     {
         std::cout << "Error: request length " << bytes << "is less than specified in the request - " << length << "\n";
```

Comment:
In case of exception the server stops working - you do not call `startReceive`. The server must keep working as much as possible. A single errorneous request is not a reason to give up :)